cmd/local-gadget/containers/containers.go

```diff
@@ -127,65 +92,30 @@ func NewListContainersCmd() *cobra.Command {
 		"Don't truncate container ID",
 	)
 
-	utils.AddCommonFlags(cmd, commonFlags)
+	utils.AddCommonFlags(cmd, &commonFlags)
 
 	return cmd
 }
 
-func (p *ContainerParser) SortContainers(containers []*containercollection.Container) {
-	sort.Slice(containers, func(i, j int) bool {
-		si, sj := (containers)[i], (containers)[j]
-		switch {
-		case si.Runtime != sj.Runtime:
-			return si.Runtime < sj.Runtime
-		case si.Name != sj.Name:
-			return si.Name < sj.Name
-		default:
-			return si.ID < sj.ID
-		}
-	})
-}
+func newContainerParser(
+	outputConfig *commonutils.OutputConfig, flags *ContainerFlags,
+) (
+	*commonutils.GadgetParser[containercollection.Container], error,
+) {
+	containerColumns := containercollection.GetColumns()
 
-func (p *ContainerParser) TransformToColumns(c *containercollection.Container) string {
-	var sb strings.Builder
-
-	for _, col := range p.OutputConfig.CustomColumns {
-		switch col {
-		case "runtime":
-			sb.WriteString(fmt.Sprintf("%s", c.Runtime))
-		case "id":
-			if p.containerFlags.noTrunc {
-				sb.WriteString(fmt.Sprintf("%s", c.ID))
-			} else {
-				sb.WriteString(fmt.Sprintf("%.13s", c.ID))
-			}
-		case "name":
-			sb.WriteString(fmt.Sprintf("%s", c.Name))
-		case "pid":
-			sb.WriteString(fmt.Sprintf("%d", c.Pid))
-		case "mntns":
-			sb.WriteString(fmt.Sprintf("%d", c.Mntns))
-		case "netns":
-			sb.WriteString(fmt.Sprintf("%d", c.Netns))
-		case "namespace":
-			sb.WriteString(fmt.Sprintf("%s", c.Namespace))
-		case "podname":
-			sb.WriteString(fmt.Sprintf("%s", c.Podname))
-		case "poduid":
-			sb.WriteString(fmt.Sprintf("%s", c.PodUID))
-		case "cgrouppath":
-			sb.WriteString(fmt.Sprintf("%s", c.CgroupPath))
-		case "cgroupid":
-			sb.WriteString(fmt.Sprintf("%d", c.CgroupID))
-		case "cgroupv1":
-			sb.WriteString(fmt.Sprintf("%s", c.CgroupV1))
-		case "cgroupv2":
-			sb.WriteString(fmt.Sprintf("%s", c.CgroupV2))
-		default:
-			continue
-		}
-		sb.WriteRune('\t')
+	idCol, ok := containerColumns.GetColumn("id")
+	if !ok {
+		return nil, fmt.Errorf(`"id" column doesn't exist`)
+	}
+
+	// By default, the "id" column is configured with width=13 and no fixed. In
+	// this way, the auto-scale will do its magic when --no-trunc is not set.
+	// Instead, when it is set, we must ensure auto-scale doesn't truncate it.
+	if flags.noTrunc {
+		idCol.Width = 64
+		idCol.FixedWidth = true
 	}
```

Comment:
@flyth What do you think about this approach? If you agree with this approach, maybe having an API `containerColumns.SetWidth` and `containerColumns.SetFixed` would be useful. Don't you think?

Reply:
So the result would be a still consistent table (headers and rows look uniformly) with the id field set to 64? But what about other fields that we don't know the maximum width of? They'd still get truncated, right?
Regarding the API: do you mean it should look like `containerColumns.SetWidth(columnName, width)` and `containerColumns.SetFixed(columnName, fixedBool)` and then mark the auto-scaler as dirty to recalculate?

Maybe instead of setting different widths for columns in case of `noTrunc`, we can use an option for the formatter telling it to use the maxWidths (fixed) instead of the normal widths. This doesn't solve the issue with the fields we don't know the limit of, but it avoids extra code.

Reply:
Btw. - as you're printing a list there and all rows are known beforehand: you can call `formatter.AdjustWidthsToContent(entries, false, 0, false)` and it should adjust the widths automatically.

Reply:
> So the result would be a still consistent table (headers and rows look uniformly) with the id field set to 64? 
>
Yes, it would be a consistent table, as shown in the PR description. 

> But what about other fields that we don't know the maximum width of? They'd still get truncated, right? 
>
Yes, they will still be truncated. The `--no-trunc` flag [applies only to the ID field](https://github.com/kinvolk/inspektor-gadget/blob/81a9ce5a2584e7bd17c8c8b1b7166f9cf084a1a6/cmd/local-gadget/containers/containers.go#L123-L128). It is the same as `docker ps --no-trunc`.

> Regarding the API: do you mean it should look like containerColumns.SetWidth(columnName, width) and containerColumns.SetFixed(columnName, fixedBool) and then mark the auto-scaler as dirty to recalculate?
> 
Is it mandatory to mark auto-scale as dirty? It is my point. If I modify those values manually and everything works fine (as it seems), then it is ok for me to export them and allow users to modify them. However, if it is needed to do any additional step, like mark it as dirty, to make the auto-scale feature work correctly, then I think we should have a helper.

> Maybe instead of setting different widths for columns in case of noTrunc, we can use an option for the formatter telling it to use the maxWidths (fixed) instead of the normal widths. This doesn't solve the issue with the fields we don't know the limit of, but it avoids extra code.
> 
I am unsure how it applies in this case. Maybe you thought about it when you didn't know we wanted to avoid truncating only one column.

Reply:
> Btw. - as you're printing a list there and all rows are known beforehand: you can call `formatter.AdjustWidthsToContent(entries, false, 0, false)` and it should adjust the widths automatically.
>
I completely forgot about this feature. I will apply it.

Reply:
> It is the same as docker ps --no-trunc.
Ah, I missed that part. You're right then a global "use maxWidth" won't apply.

> Is it mandatory to mark auto-scale as dirty?
Yes. Unless you change things before the autoscaler get's called for the first time (which happens when you output a header or call it manually). So either a SetDirty() or a couple of Setters for the formatter - I'm fine with both.

Reply:
> > Is it mandatory to mark auto-scale as dirty?
> 
> Yes. Unless you change things before the autoscaler get's called for the first time (which happens when you output a header or call it manually). So either a SetDirty() or a couple of Setters for the formatter - I'm fine with both.
>
OK, the current code is organized perform these operations before calling `NewGadgetParser` (who calls `NewFormatter`). That's the reason it is working and should keep working. We can keep it like this for now.

On the other hand, I tried `AdjustWidthsToContent`, but let's resolve this discussion and start another.